metadata_backend/api/operators.py

```diff
@@ -677,3 +677,119 @@ def _generate_folder_id(self) -> str:
         sequence = "".join(secrets.choice(string.digits) for i in range(8))
         LOG.debug("Generated folder ID.")
         return f"FOL{sequence}"
+
+
+class UserOperator:
+    """Operator class for handling database operations of users.
+
+    Operations are implemented with JSON format.
+    """
+
+    def __init__(self, db_client: AsyncIOMotorClient) -> None:
+        """Init db_service.
+
+        :param db_client: Motor client used for database connections. Should be
+        running on same loop with aiohttp, so needs to be passed from aiohttp
+        Application.
+        """
+        self.db_service = DBService("users", db_client)
+
+    async def create_user(self, data: Dict) -> str:
+        """Create new user object to database.
+
+        :param data: Data to be saved to database
+        :raises: 400 if error occurs during the process
+        :returns: User id for the user object inserted to database
+        """
+        user_id = self._generate_user_id()
+        data["userId"] = user_id
+        try:
+            insert_success = await self.db_service.create("user", data)
+        except (ConnectionFailure, OperationFailure) as error:
+            reason = f"Error happened while inserting user: {error}"
+            LOG.error(reason)
+            raise web.HTTPBadRequest(reason=reason)
+
+        if not insert_success:
+            reason = "Inserting user to database failed for some reason."
+            LOG.error(reason)
+            raise web.HTTPBadRequest(reason=reason)
+        else:
+            LOG.info(f"Inserting user with id {user_id} to database succeeded.")
+            return user_id
+
+    async def read_user(self, user_id: str) -> Dict:
+        """Read user object from database.
+
+        :param user_id: User ID of the object to read
+        :raises: 400 if reading was not successful
+        :returns: User object formatted to JSON
+        """
+        await self._check_user_exists(self.db_service, user_id)
+        try:
+            user = await self.db_service.read("user", user_id)
+        except (ConnectionFailure, OperationFailure) as error:
+            reason = f"Error happened while getting user: {error}"
+            LOG.error(reason)
+            raise web.HTTPBadRequest(reason=reason)
+        return user
+
+    async def update_user(self, user_id: str, data: JsonPatch) -> str:
+        """Update user object from database.
+
+        :param user_id: ID of user to update
+        :param data: Data to be updated
+        :returns: ID of the user updated to database
+        """
+        await self._check_user_exists(self.db_service, user_id)
+        try:
+            update_success = await self.db_service.update("user", user_id, data)
```

Comment:
I'd suggest you to write a json schema for users too and do following checks 

 1. Don't let patch to modify user data coming from external sources (probably name, id etc.), only folders and drafts for now.
2. Do validation for updated content like folders and optionally sanity check too after saving updated content to db